pkg/cloud/services/networking/floatingip.go

```diff
@@ -122,8 +122,13 @@ var backoff = wait.Backoff{
 func (s *Service) AssociateFloatingIP(eventObject runtime.Object, fp *floatingips.FloatingIP, portID string) error {
 	s.scope.Logger.Info("Associating floating IP", "id", fp.ID, "ip", fp.FloatingIP)
 
+	if len(fp.PortID) != 0 {
+		s.scope.Logger.Info("Floating IP already associated to other port:", "id", fp.ID, "fixed ip", fp.FixedIP)
+		return nil
+	}
+
 	if fp.PortID == portID {
```

Comment:
Can this code be reached?
If checked for != 0 in line 125, line 130 can only be reached if `fp.PortID` is empty.

Reply:
you are right, so we only need check the len is 0 or not on such logic, thanks~